src/transformers/models/bridgetower/configuration_bridgetower.py

```diff
@@ -0,0 +1,403 @@
+# coding=utf-8
+# Copyright 2022 The Intel Labs Team Authors, The Microsoft Research Team Authors and HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License=, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing=, software
+# distributed under the License is distributed on an "AS IS" BASIS=,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND=, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" BridgeTower model configuration"""
+
+import copy
+import os
+from typing import Union
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+BRIDGETOWER_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "BridgeTower/bridgetower-base": "https://huggingface.co/BridgeTower/bridgetower-base/blob/main/config.json",
+    "BridgeTower/bridgetower-base-itm-mlm": (
+        "https://huggingface.co/BridgeTower/bridgetower-base-itm-mlm/blob/main/config.json"
+    ),
+}
+
+
+class BridgeTowerVisionConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the vision configuration of a [`BridgeTowerModel`]. Instantiating a
+    configuration with the defaults will yield a similar configuration to that of the bridgetower-base
+    [BridegTower/bridgetower-base](https://huggingface.co/BridgeTower/bridgetower-base/) architecture.
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+    Args:
+        output_resolution (`int`, *optional*, defaults to 288):
+            The final size (resolution) of each image.
+        input_resolution_before (`int`, *optional*, defaults to 224):
+            The size (resolution) of each input image.
+        stop_gradient (`bool`, *optional*, defaults to `False`):
+            Whether to stop gradient for training.
+        embed_dim (`int`, *optional*, defaults to 512):
+            Dimension of embeddings in vit model.
+        width (`int`, *optional*, defaults to 768):
+            Dimensionality of the encoder layers and the pooler layer.
+        layers (`int`, *optional*, defaults to 12):
+            Number of hidden layers in vit model.
+        vit_layernorm_init_from_vit (`bool`, *optional*, defaults to `False`):
+            Whether to init vit LayerNorm from vit.
+        vit_layernorm_shared (`bool`, *optional*, defaults to `True`):
```

Comment:
```suggestion
        share_layernorm (`bool`, *optional*, defaults to `True`):
```
Can't we also remove ViT from the names since this is the vision config anyway?

Reply:
Fixed this. Thanks for the suggestion!